examples/nlu-assistant/README.md

```diff
@@ -0,0 +1,515 @@
+
+##### 2.4 Using your own custom model
+
+Now, let's move again to **src/nlu/train-model.ts**.
+
+###### 2.4.1 Initializing the NLU with custom preprocessors
+
+**src/nlu/train-model.ts**
+
+```ts
+import { BotonicNLU } from '@botonic/nlu'
+import { join } from 'path'
+import { train } from '@tensorflow/tfjs-node'
+import { DenseLayerArgs } from '@tensorflow/tfjs-layers/dist/layers/core'
+import { LSTMLayerArgs, ModelCompileArgs } from '@tensorflow/tfjs-layers/dist'
+
+import { tokenizer } from './preprocessing-tools/tokenizer'
+import { DefaultNormalizer } from './preprocessing-tools/normalizer'
+import { LSTMModel } from './lstm-model'
+import { readdirSync } from 'fs'
+
+const LANGUAGE = 'en'
+const UTTERANCES_PATH = join(
+  process.cwd(),
+  'src',
+  'nlu',
+  'utterances',
+  LANGUAGE
+)
+const MAX_SEQ_LEN = 20
+const INTENTS_COUNT = readdirSync(UTTERANCES_PATH).length
+
+const nlu = new BotonicNLU({
+  tokenizer: tokenizer,
+  normalizer: new DefaultNormalizer(),
+})
+```
+
+Call `BotonicNLU` constructor and initialize it by passing your imported tokenizer and normalizer.
+
+###### 2.4.2 Preparing the data
+
+**src/nlu/train-model.ts** (continuation)
+
+```ts
+const data = nlu.readData({
+  path: UTTERANCES_PATH,
+  language: LANGUAGE,
+  maxSeqLen: MAX_SEQ_LEN,
+})
+const [xTrain, xTest, yTrain, yTest] = nlu.trainTestSplit({
+  data: data,
+  testPercentage: 0.15,
+  stratify: false,
+})
+```
+
+At this point, you only had to call `readData` and all the preprocessing will be handled internally with the preprocesors you have injected before. As we have an homogenized distribution of the data (same number of examples per intent), we will be setting `stratify` option to `false`.
+
+###### 2.4.3 Model configuration and tunning
+
+**src/nlu/train-model.ts** (continuation)
+
+```ts
+const WE_CONFIG = {
+  language: LANGUAGE,
+  dimension: 50,
+  type: 'glove',
+  vocabulary: nlu.vocabulary,
+} as WordEmbeddingsConfig
+
+const LAYERS_CONFIG = {
+  LSTM: {
+    units: 32,
+    dropout: 0.4,
+    recurrentDropout: 0.4,
+  } as LSTMLayerArgs,
+  DENSE: {
+    units: INTENTS_COUNT,
+    activation: 'softmax',
+  } as DenseLayerArgs,
+}
+```
+
+Time to configure our model.
+
+- **Word Embeddings**:
+  - Here we have chosen [`glove`](https://nlp.stanford.edu/projects/glove/) `50d` pretrained word embeddings. We must pass here the vocabulary extracted from previous data processing.
+- **LSTM Layer**:
+  - As we said before, we want our model to be less rigid when predicting over new inputs. To achive that, we use a technique called `regularization` that can be done by specifying `dropout` parameters to this layer. Also, reducing the number of `units` (neurons per layer) will help.
+- **Dense Layer**:
+  - This is the very last layer and **must** have as many units as intents we are targeting. Since we have multiple intents (classes) we have to make use of `softmax` activation.
```

Comment:
The softmax activation function is not used because we have several intents. It is used to compute the probabilities of all intents so the sum of all of them is equal to 1.